BooruSharp.Others/Pixiv.cs

```diff
@@ -200,6 +202,60 @@ public override async Task<SearchResult[]> GetLastPostsAsync(params string[] tag
             return ParseSearchResults((JArray)json["illusts"]);
         }
 
+        public async Task<SearchResult[]> GetRankingAsync(RankingMode mode = RankingMode.Daily, int offset = 0, DateTime? date = null)
+        {
+            if (AccessToken == null)
+                throw new AuthentificationRequired();
+            await CheckUpdateTokenAsync();
+            string url = _baseUrl + "/v1/illust/ranking?mode=" + rankingModeValues[mode];
+            if (offset != 0)
+                url += "&offset=" + offset;
+            if (date.HasValue)
+                url += "&date=" + date.Value.ToString("yyyy-MM-dd");
+            var request = new HttpRequestMessage(HttpMethod.Get, url);
+            request.Headers.Add("Authorization", "Bearer " + AccessToken);
+
+            var http = await HttpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+            if (!http.IsSuccessStatusCode)
+                throw new Exception(http.StatusCode.ToString()); //ToDo: Implement exception
+            JToken json = (JToken)JsonConvert.DeserializeObject(await http.Content.ReadAsStringAsync());
+            return ParseSearchResults((JArray)json["illusts"]);
+        }
+
+        public enum RankingMode
+        {
+            Daily = 0,
+            Weekly = 1,
+            Monthly = 2,
+            Popular_among_male_users = 3,
+            Popular_among_female_users = 4,
+            Original = 5,
+            Manga_Rookie = 6,
+            Manga_Daily = 7,
+            R18_Daily = 8,
+            R18_Weekly = 9,
+            R18_Popular_among_male_users = 10,
+            R18_Popular_among_female_users = 11,
+            R18G_Weekly = 12
```

Comment:
```suggestion
            PopularAmongMaleUsers = 3,
            PopularAmongFemaleUsers = 4,
            Original = 5,
            MangaRookie = 6,
            MangaDaily = 7,
            R18Daily = 8,
            R18Weekly = 9,
            R18PopularAmongMaleUsers = 10,
            R18PopularAmongFemaleUsers = 11,
            R18GWeekly = 12
```
As per [Microsoft guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions#capitalization-rules-for-identifiers), PascalCase is the preferred way of naming enum values.
Also explicitly specifying the integer values isn't necessary.